alerter.go

```diff
@@ -111,36 +111,43 @@ func ReportErrors(config Config, errors []string) []string {
 	return report
 }
 
-func included(logMessage string, includes []string) bool {
-	if len(includes) == 0 {
-		return true
-	}
-
-	for _, include := range includes {
-		matched, err := regexp.MatchString(include, logMessage)
-		if err != nil {
-			panic(err)
-		}
+type PatternMatcher struct {
+	includes []*regexp.Regexp
+	excludes []*regexp.Regexp
+}
 
-		if matched {
-			return true
+func NewPatternMatcher(includes []string, excludes []string) *PatternMatcher {
+	toRegexp := func(patterns []string) []*regexp.Regexp {
+		allRegexp := make([]*regexp.Regexp, 0, len(patterns))
+		for _, pattern := range patterns {
+			allRegexp = append(allRegexp, regexp.MustCompile(pattern))
 		}
+		return allRegexp
 	}
-	return false
+	return &PatternMatcher{includes: toRegexp(includes), excludes: toRegexp(excludes)}
 }
 
-func excluded(logMessage string, excludes []string) bool {
-	for _, exclude := range excludes {
-		matched, err := regexp.MatchString(exclude, logMessage)
-		if err != nil {
-			panic(err)
+func (matcher *PatternMatcher) FindAllMatch(s string) []string {
+	matches := make([]string, 0, 0)
+	for _, include := range matcher.includes {
+		for _, match := range include.FindAllString(s, -1) {
+			matches = append(matches, match)
 		}
+	}
 
-		if matched {
-			return true
+	for _, exclude := range matcher.excludes {
+		excludedCount := 0
+		for i := range matches {
+			j := i - excludedCount
+			match := matches[j] // ensure we get the correct element taking deletes into account
+
+			if len(exclude.FindAllString(match, -1)) != 0 {
+				matches = append(matches[:j], matches[j+1:]...)
+				excludedCount++
+			}
 		}
 	}
-	return false
+	return matches
```

Comment:
The exclusion is working now, but i think it is a bit complicated to understand what is going on. 
Unless we are doing this to avoid memory issues by creating a new slice with only the accepted elements, then i would suggest something like https://play.golang.org/p/S2xcmxgORQ.
But I am happy to go with either.